client/finality-grandpa-warp-sync/src/proof.rs

```diff
@@ -29,8 +29,8 @@ use sp_runtime::{
 
 use crate::HandleRequestError;
 
-/// The maximum number of authority set change proofs to include in a single warp sync proof.
-const MAX_CHANGES_PER_WARP_SYNC_PROOF: usize = 256;
+/// The maximum size in bytes of the `WarpSyncProof`.
+pub(super) const MAX_WARP_SYNC_PROOF_SIZE: usize = 16 * 1024 * 1024;
```

Comment:
Should we not use value that is a little bit lower? Or is the 16MiB limit directly for the message content?

Reply:
I'm not sure I get your question, but yes 16MiB is the maximum size we want to send on the network at once.
It means that the remote might have to allocate up to 16MiB of memory to buffer the message, not more.

Reply:
In theory I guess that we don't necessarily need a limit, and we could somehow stream the response. The remote could use yamux to accept the data progressively, and verify the justifications in the order they are received. But it's so much easier to just buffer everything and process everything at once.

Reply:
What I wanted to say/ask. We are probably wrapping the messages until we end up with the "final" message format that we send over the socket. And wrapping also means we consume some more bytes, because we store extra data in the message etc. 

So, I wanted to know to at which state do we apply the limit of 16MiB?

Reply:
No, the proof is exactly the message being sent.